components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImplTestCase.java

```diff
@@ -558,17 +559,23 @@ public void testUpdateNullAPI() throws APIManagementException {
     @Test(description = "Update api status")
     public void testUpdateAPIStatus() throws APIManagementException, LifecycleException {
         ApiDAO apiDAO = Mockito.mock(ApiDAO.class);
+        WorkflowDAO workflowDAO = Mockito.mock(WorkflowDAO.class);
         APILifecycleManager apiLifecycleManager = Mockito.mock(APILifecycleManager.class);
         APIPublisherImpl apiPublisher = new APIPublisherImpl(user, apiDAO, null, null, null, apiLifecycleManager, null,
```

Comment:
Very recently we introduced private methods to retrieve Impl objects as in - https://github.com/pubudu538/carbon-apimgt/blob/a554f8712c4cc31ea223f3d94fab7ad19f4f7cdc/components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImplTestCase.java#L123. With this effort we can avoid changing all the test cases when adding a new DAO layer.  Shall we change this?